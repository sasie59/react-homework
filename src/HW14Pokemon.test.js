import React from "react";
import { render, waitFor } from "@testing-library/react";
import "@babel/polyfill";
import HW from "./HW14Pokemon";

global.fetch = jest.fn(
  () =>
    new Promise((resolve) => {
      resolve({
        json: () => new Promise((resolve) => {
          resolve({
            res: [
              {
                count: 1118,
                next: "https://pokeapi.co/api/v2/pokemon/?offset=100&limit=3",
                previous: null,
                results: [
                  { name: "bulbasaur", url: "https://pokeapi.co/api/v2/pokemon/1/" },
                  { name: "ivysaur", url: "https://pokeapi.co/api/v2/pokemon/2/" },
                  { name: "venusaur", url: "https://pokeapi.co/api/v2/pokemon/3/" },
                ],
              },
            ],
          });
        }),
      });
    })
);

/**  */
test("should HW14Pokemon", () => {
  expect(true).toBeTruthy();
});

/** 根據我GOOGLE，我發現只要有 fetch 的測試，都要使用 await / async 去接測試，像這樣 */
// test("should HW14Pokemon", async () => {
//   const { container } = render(<HW />);
//   await waitFor(() => expect(container).toMatchSnapshot());
// });