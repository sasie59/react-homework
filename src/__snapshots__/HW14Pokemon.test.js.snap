// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should HW14Pokemon 1`] = `
<div>
  <div>
    <h2>
      順利抓取 100 隻寶可夢的名字與圖片
    </h2>
    <h2>
      可以依名字篩選出寶可夢
    </h2>
    <span>
      <b>
        keyword
      </b>
    </span>
    <input
      placeholder="name..."
      type="text"
      value=""
    />
  </div>
</div>
`;
